### ISSUE TITLE
Calendar: Inline violation of No Keyboard Trap -  WCAG SC 2.1.2

### ISSUE THREAD

**Issue by harrisonwillettbetterworks** [primefaces/primevue#4055]
### Describe the bug

The Calendar component has a WCAG SC 2.1.2 failure when in "inline" mode found here:
[/components/lib/calendar/Calendar.vue](https://github.com/primefaces/primevue/blob/master/components/lib/calendar/Calendar.vue)

### Reproducer

https://primevue.org/calendar/#inline

### PrimeVue version

3.29.2

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

Chrome Version 114.0.5735.106

### Steps to reproduce the behavior

1. Given you are using an "inline" "calendar" as seen here:
https://primevue.org/calendar/#inline
2. When using "Shift" + "Tab" on the first element in the calendar the user is taken to the last element in the calendar which would be correct for the dialog mode but not for inline mode as the user can not get to the previously focusable element.

### Expected behavior

Keep existing behavior for dialogs but when in the inline mode pressing  "Shift" + "Tab" on the first element in the calendar should result in visiting the previous focusable element and leave the calendar widget.
